### ISSUE TITLE
Move axum_extra's `Multipart` into axum

### ISSUE THREAD

**Issue by davidpdrsn** [tokio-rs/axum#1952]
axum extra has its own `Multipart` type which does some borrow checking at runtime, making it possible to implement common and safe use cases. See https://github.com/tokio-rs/axum/pull/1692 for more context.

For 0.7 we should consider if we should promote that to axum.

**Comment by mo8it** [comment]
Any plans on providing something like [typed-multipart](https://github.com/murar8/axum_typed_multipart)?

**Comment by davidpdrsn** [comment]
That crate seems to already provide it. I have no plans of adding that to axum itself.

**Comment by rakshith-ravi** [comment]
@davidpdrsn is there a reason you're considering to move it into axum? Having it in axum-extra seems fine.

On the flip side, having it in axum would certainly increase the amount of features we provide out of the box, making axum more "batteries-included" (at least more than before). That being said, if you are moving it into core axum, I'd strongly urge you to feature gate it so that it doesn't affect compile times.

**Comment by davidpdrsn** [comment]
axum has less breaking releases than axum-extra so it more about maturity and long term stability.

I still think it’s appropriate for axum to provide a multipart extractor.

**Comment by rakshith-ravi** [comment]
Ahh, got it. We could perhaps put out some sort of a "community feedback wanted" thingy where we understand how many people use the multipart extractor in production (I personally don't so I'm not able to give enough feedback) and if they face any issues, both in terms of bugs as well as DX. If what's currently there seems good, then we know that we have something that's good to go and wouldn't break a lot, and can move it into axum I suppose. Thoughts?

**Comment by jplatte** [comment]
I haven't seen a lot of demand for this and personally I like the compile-time safety of the current multipart implementation in `axum`, so I'll remove this from the 0.8 milestone.

**Comment by davidpdrsn** [comment]
I agree that borrow checking is nicer. I'll close this for now since there doesn't appear to be much demand judging by the level of activity. But if someone still really wants this we can re-open.
